src/librustdoc/Cargo.toml

```diff
@@ -3,10 +3,6 @@ authors = ["The Rust Project Developers"]
 name = "rustdoc"
 version = "0.0.0"
 
-[lib]
-name = "rustdoc"
-path = "lib.rs"
-
```

Comment:
Why move all the files into an inner `src` directory? Does anything else in-tree do this?

Reply:
Because it's the default.

>  Does anything else in-tree do this?

I'm not sure, but I'm of the opinion it should ;)

Reply:
If it weren't for the error index generator using bits of librustdoc, we may be able to get away with moving all this code into `src/tools/rustdoc`, where this pattern would definitely fit in. As-is, nothing else directly in `src/` is structured like this. I'm personally indifferent to the change, but it's going to break (and be broken by) every rustdoc PR. (And several people seem interested in cleaning up rustdoc right now...)

Heck, we could possibly get away with it anyway, since a Cargo project can have both a lib and a bin. We'd just need to point bootstrap and the error index generator (and anything else that may point at librustdoc directly) at the new folder.

Reply:
I think it's better to keep it. At least some tests rely on it (like error index rendering iirc).

Reply:
If we're moving all the source files we should at least rename `librustdoc` to just `rustdoc`.